source/extensions/common/wasm/wasm.cc

```diff
@@ -539,11 +539,7 @@ Word ___syscall146Handler(void*, Word, Word) { throw WasmException("emscripten s
 
 void ___setErrNoHandler(void*, Word) { throw WasmException("emscripten setErrNo"); }
 
-// NB: pthread_equal is required to return 0 by the protobuf libarary.
-Word _pthread_equalHandler(void*, Word,
-                           Word) { /* throw WasmException("emscripten pthread_equal"); */
-  return 0;
-}
+Word _pthread_equalHandler(void*, Word left, Word right) { return left == right; }
```

Comment:
This was during testing with wasm-sb BTW.

Reply:
> Given that we don't support pthread_create, this should always be true since there should be no way to create another thread.

Good point, but then it should be always returning `1` (equal), not `0` (not equal).

WASM calls `pthread_equal` to compare `thread_id` with either the same `thread_id` or `0`, so this implementation sometimes returns `0` (not equal) and sometimes returns `1` (equal), and `wasm-sd` seems to be fine with it.

For the record: hardcoding return value to `0` (not equal) seems to work fine as well, and hardcoding return value to `1` (equal) results in early failure during `wasm-sd` startup.